### ISSUE TITLE
"return-undefined" requires explicit "return undefined" even for functions contextually-typed "any"

### ISSUE THREAD

**Issue by Kovensky** [palantir/tslint#2566]
### Bug Report

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.3.0
- __Running TSLint via__: tslint-loader/webpack

#### TypeScript code being linted

```ts
$('.play-button').on('click', e => {
  if (playing) {
    return
  }
  // rest of function
})
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint-config-standard"
  ]
}
```

#### Actual behavior

Error on the `return`, requiring that it be a `return undefined`

#### Expected behavior

This function is a void function that just happens to be contextually typed to return `any` by the jquery type definitions; it should not be required to explicitly return `undefined`, especially not if the contextual type is `any`.
